src/server/mod.rs

```diff
@@ -135,6 +136,7 @@ fn find_references<'a>(
 #[derive(Default)]
 struct LspServerState {
     buckets: Vec<String>,
+    composition: Option<RefCell<composition::Composition>>,
```

Comment:
I definitely don't want use to use server state to store this. We have the mechanisms to parse it from the ast each time, and we should use that to get the composition state. On top of that, composition is file-level state, not server level, and the source of truth for that state is always the client. In this case, we have a place where that state may get out of sync.

Reply:
Agreed. I took the stateful nature of composition, as a desiring a stateful LspServer for composition.

Prefer your route. Updates made and pushed.